Update QubesDB entries of given device (and its partitions) when it's attached
or detached to other VM. This is used to prevent accidential attachment of
whole device and its partitions at the same time.
--- xen-4.6.1/tools/hotplug/Linux/block.orig	2016-05-16 03:38:46.950000000 +0200
+++ xen-4.6.1/tools/hotplug/Linux/block	2016-05-16 03:52:27.934000000 +0200
@@ -248,6 +248,12 @@
         check_device_sharing "$dev" "$mode"
 	write_dev "$dev"
         release_lock "block"
+        # update QubesDB entries
+        if [ -f "/sys/class/block/$(basename $dev)/partition" ]; then
+            udevadm trigger --property-match=DEVNAME=$dev
+        else
+            udevadm trigger --parent-match=/sys/class/block/$(basename $dev)
+        fi
 	exit 0
 	;;
 
@@ -330,8 +336,20 @@
 
   remove)
     truetype=$(cat $HOTPLUG_STORE-type || echo $truetype)
+    p=$(cat $HOTPLUG_STORE-params || echo $p)
     case $truetype in
       phy)
+        dev=$(expand_dev $p)
+        if [ -L "$dev" ]
+        then
+          dev=$(readlink -f "$dev") || fatal "$dev link does not exist."
+        fi
+        # update QubesDB entries
+        if [ -f "/sys/class/block/$(basename $dev)/partition" ]; then
+            udevadm trigger --property-match=DEVNAME=$dev
+        else
+            udevadm trigger --parent-match=/sys/class/block/$(basename $dev)
+        fi
 	exit 0
 	;;
 
